base-template.sh

```diff
@@ -45,6 +45,24 @@ are_githooks_disabled() {
     return 1
 }
 
+#####################################################
+# Sets the ${INSTALL_DIR} variable
+
+# Returns: 0 if success, 1 otherwise
+#####################################################
+set_install_dir() {
+    INSTALL_DIR=$(git config --global githooks.installDir)
+
```

Comment:
Please change this (and also in `cli.sh`) to something like this:
```
load_install_dir() {
    INSTALL_DIR="$(git config --global githooks.installDir)"
    if [ -n "${INSTALL_DIR}" ]; then
        if [ -d "${INSTALL_DIR}" ]; then
            return
        else
            echo "! Githooks installation is corrupt! " >&2
            echo "  Install directory ${INSTALL_DIR} is missing." >&2
            echo "  Please run the Githooks install script again to fix it." >&2
        fi
    fi

    INSTALL_DIR="~/.githooks"
}
```

Reply:
You really want to default here to "~/.githooks". Why?
I dont think this is a good idea, just very prone to undefined behavior...

Reply:
Yes, I want to make sure that the previous behaviour is not broken, and that sensible defaults are applied for users who don't care about where things are installed.

Reply:
ok you printed a warning...

Reply:
What happens when someone upgrades a single install repo?
I'm worried about this change, because someone might upgrade from an older version, not get the new config (for whatever reason), then we'd print this error message.
I think when we don't have a value set, we should silently fall back to the default value.
If the config is set, but the directory it points to doesn't exist, then we should print an error.

On a side note: are you not doing this to make Githooks working nicely in both Git Bash and WSL at the same time? Will you even have the same paths there?